utils/src/deployer.rs

```diff
@@ -399,6 +418,25 @@ fn should_deploy(group: ContractGroup, only: &Option<Vec<ContractGroup>>) -> boo
     }
 }
 
+pub async fn is_proxy_contract(
```

Comment:
this looks good to me. We can use it to validate each contract address in toml file  (base + upgrades) by adding validate() to Genesis.

Reply:
https://github.com/EspressoSystems/espresso-sequencer/pull/2000#discussion_r1763274927